lib/rspec/core/option_parser.rb

```diff
@@ -291,6 +291,8 @@ def initialize_project_and_exit
     def bisect_and_exit(argument)
       RSpec::Support.require_rspec_core "bisect/coordinator"
 
+      # This might be the best spot to mutate ENV['SPEC_OPTS']
+      # if we do that.
```

Comment:
> Surely a single instance per process...

A `Bisect::Runner` instance is only created in the main bisect process, not in the processes created to run subsets of the suite.  So yes, a single instance per main bisect process, but there's only one main bisect process for an entire `--bisect` run.

It's made here in the coordinator:

https://github.com/rspec/rspec-core/blob/cc869c8b8c1e00991b8be0c71faa9a5a74b8fd22/lib/rspec/core/bisect/coordinator.rb#L33

...and is then used repeatedly in the example minimizer to run specific example subsets:

https://github.com/rspec/rspec-core/blob/cc869c8b8c1e00991b8be0c71faa9a5a74b8fd22/lib/rspec/core/bisect/example_minimizer.rb#L142